apriltag_ros/src/continuous_detector.cpp

```diff
@@ -105,9 +187,29 @@ void ContinuousDetector::imageCallback (
   // Check for an empty image and publish an empty detection message
   if (image_rect->data.empty())
   {
-    AprilTagDetectionArray tag_detection_array;
-    tag_detection_array.header = image_rect->header;
-    tag_detections_publisher_.publish(tag_detection_array);
+    publishEmptyDetection(image_rect->header);
+    return;
+  }
+
+  // Transform the tag pose from robot's base frame to optical frame
+  if (!toOptical(robot_pose_header_, image_rect->header.frame_id, tag_pose_))
+  {
+    publishEmptyDetection(image_rect->header);
+    return;
+  }
+
+  // Set the camera info
+  camera_model_.fromCameraInfo(camera_info);
+  // compute the pixel values for the FOV pixel buffer (ie. resizing the FOV)
+  fov_pixel_buffer_height_ = (camera_model_.reducedResolution().height/2 
+    - (camera_model_.reducedResolution().height * fov_size_scaler_) / 2);
+  fov_pixel_buffer_width_ = (camera_model_.reducedResolution().width/2 
+    - (camera_model_.reducedResolution().width * fov_size_scaler_) / 2);
```

Comment:
I'd use local variables and maybe also another variable for the scale factor:
```suggestion
  const auto height = camera_model_.reducedResolution().height;
  const auto width = camera_model_.reducedResolution().width;

  const auto fov_scale = 0.5 * (1.0 - fov_size_scaler_);

  fov_pixel_buffer_height_ = height * fov_scale;
  fov_pixel_buffer_width_ = width * fov_scale;
```

Reply:
or store reducedResolution() in a variable, since that call actually computes things, its not just a private variable getter

http://docs.ros.org/en/diamondback/api/image_geometry/html/c++/pinhole__camera__model_8cpp_source.html#l00175

Reply:
changed.